### PR TITLE
Make User Abstract Class

### PR DESCRIPTION
**Pull request by manya-mittal**
Please review the Abstract User Class that I coded

### CODE REVIEW

File: src/main/java/Entities/User.java [hunk 1]
```diff
@@ -1,4 +1,20 @@
 package Entities;
 
-public interface User {
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class User {
+    private String Username;
+    private String password;
+    List<Playlist> playlistList = new ArrayList<Playlist>();
```
Comment (sarxhlwalker) [inline src/main/java/Entities/User.java:9]: instance variable should have a private/public attribute

**Review by sarxhlwalker — CHANGES_REQUESTED** [review]
Please see comments.

File: src/main/java/Entities/User.java [hunk 2]
```diff
@@ -1,4 +1,20 @@
 package Entities;
 
-public interface User {
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class User {
+    private String Username;
+    private String password;
```
Comment (chrstinalin) [inline src/main/java/Entities/User.java:8]: Capitalization on Username
Kind of redundant but you /could/ make this one line? 

private String username, password;

**Review by chrstinalin — CHANGES_REQUESTED** [review]
Very minor issues, otherwise good

**Review by chrstinalin — DISMISSED** [review]
Looks good to me :)

File: src/main/java/Entities/User.java (hunk 2 above)
Comment (rafaelgacesa) [reply src/main/java/Entities/User.java]: One liners can often lead to code which is a lot more difficult to debug; remember that primarily you write code to be read by other developers (maybe many months after you create the initial version).

The compiler doesn’t need one liners - in fact the good C and C++ compilers for instance will remove temporary variables and optimize the code far better than you can by writing one-liners.

**Review by sarxhlwalker — DISMISSED** [review]
Wow, gorgeous code!

File: src/main/java/Entities/User.java (hunk 2 above)
Comment (manya-mittal) [reply src/main/java/Entities/User.java]: Thanks for your feedback Christina! I agree with Rafael, I think code in one line might be more difficult to read and understand later

File: src/main/java/Entities/User.java [hunk 3]
```diff
@@ -1,4 +1,24 @@
 package Entities;
 
-public interface User {
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class User {
+    private final String username;
+    private final String password;
+    private final List<Playlist> playlistList;
```
Comment (SiqiZhang6) [inline src/main/java/Entities/User.java:9]: delete plz

**Review by SiqiZhang6 — CHANGES_REQUESTED** [review]

**Review by rafaelgacesa — APPROVED** [review]
I did NOT ask to be here. However,  it was a pleasure to read. Thanks!

**Review by sarxhlwalker — APPROVED** [review]
Wow, gorgeous code! Please merge this PR so I can stop looking at it.

**Pull request merged**